staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -220,26 +220,18 @@ func Forbidden(field *Path, detail string) *Error {
 	return &Error{ErrorTypeForbidden, field.String(), "", detail}
 }
 
-// TooLong returns a *Error indicating "too long".  This is used to
-// report that the given value is too long.  This is similar to
-// Invalid, but the returned error will not include the too-long
-// value.
+// TooLong returns a *Error indicating "too long".  This is used to report that
+// the given value is too long.  This is similar to Invalid, but the returned
+// error will not include the too-long value. If maxLength is negative, it will
+// be included in the message.  The value argument is not used.
 func TooLong(field *Path, value interface{}, maxLength int) *Error {
-	return &Error{ErrorTypeTooLong, field.String(), value, fmt.Sprintf("must have at most %d bytes", maxLength)}
-}
-
-// TooLongMaxLength returns a *Error indicating "too long".  This is used to
-// report that the given value is too long.  This is similar to
-// Invalid, but the returned error will not include the too-long
-// value. If maxLength is negative, no max length will be included in the message.
-func TooLongMaxLength(field *Path, value interface{}, maxLength int) *Error {
 	var msg string
 	if maxLength >= 0 {
-		msg = fmt.Sprintf("may not be longer than %d", maxLength)
+		msg = fmt.Sprintf("may not be more than %d bytes", maxLength)
 	} else {
 		msg = "value is too long"
 	}
-	return &Error{ErrorTypeTooLong, field.String(), value, msg}
+	return &Error{ErrorTypeTooLong, field.String(), "<value omitted>", msg}
```

Comment:
What difference does it make whether we store the value in the `Error` or not? In other words, where is that field used?

Reply:
The value field is not used when rendering the final error string.  We could retain the value in the struct, but there are no users of this API that don't end up returning an error string, in which case the value is not used.

I found myself questioning why we pass the value when the docs say "not used" and I don't think we need anyone else wasting time on that.

Reply:
Nit: Perhaps add a comment?

I understand your rationale, but the way it is now also led to this question.

Reply:
/hold

In case that you want to change it.

Reply:
Added a commit co commet at call-sites

Reply:
/lgtm